[c2][serialized test] Use tempfile during serialized test comparison

ajyu opened the pull request:
Summary:
TestPilot runs stress tests in parallel. These fail for serialized tests because extracting (and subsequent deletion) of binary data during the process isn't threadsafe. Extract zips into tempfile to avoid this problem.

Also remove some accidentally checked in zips of a test that we didn't end up including for now

Differential Revision: D10013682

houseroad commented:
Do we have any other file we should remove, too, something like you have done in this PR?

ajyu commented:
I'll look through to double check for any other files we should remove.

ajyu committed:
Add random suffix to output directory during comparison (#12021)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/12021

TestPilot runs stress tests in parallel. These fail for serialized tests because extracting (and subsequent deletion) of binary data during the process isn't threadsafe. Add a random suffix to extracted directory to make it much more reliable. We use numpy.random because random is seeded for TestPilot runs.

Also remove some accidentally checked in zips of a test that we didn't end up including for now.

Reviewed By: houseroad

Differential Revision: D10013682

fbshipit-source-id: 842a02ff1bbe859dfdb0496858de1a09fd1573fd
